sql/2020/22_HTTP_2/detailed_upgrade_headers.sql

```diff
@@ -0,0 +1,34 @@
+# standardSQL
+# Detailed upgrade headers for 20.04, 20.05 and 20.06
+CREATE TEMPORARY FUNCTION getUpgradeHeader(payload STRING)
+RETURNS STRING
+LANGUAGE js AS """
+  try {
+    var $ = JSON.parse(payload);
+    var headers = $.response.headers;
+    var st = headers.find(function(e) { 
+      return e['name'].toLowerCase() === 'upgrade'
+    });
+    return st['value'];
+  } catch (e) {
+    return '';
+  }
+""";
+
+SELECT 
+  client,
+  firstHtml,  
+  JSON_EXTRACT_SCALAR(payload, "$._protocol") AS protocol,
+  IF(url LIKE "https://%","https","http") AS http_or_https,
+  getUpgradeHeader(payload) AS upgrade,
+  COUNT(0) AS num_requests
```

Comment:
Upgrade header (and all response headers actually) exists in the `respOtherHeaders` column so use that (similar for other queries):

```suggestion
# standardSQL
# Detailed upgrade headers for 20.04, 20.05 and 20.06
SELECT 
  client,
  firstHtml,  
  JSON_EXTRACT_SCALAR(payload, "$._protocol") AS protocol,
  IF(url LIKE "https://%","https","http") AS http_or_https,
  regexp_extract(regexp_extract(respOtherHeaders, r'(?is)Upgrade = (.*)'), r'(?im)^([^=]*?)(?:, [a-z-]+ = .*)') is not null AS upgrade,
  COUNT(0) AS num_requests
```

Reply:
Changes merged into other changes made by @rviscomi